### ISSUE TITLE
AddRemoteFile fails because of token in querystring

### ISSUE THREAD

**Issue by jjjjeeffff** [slack-go/slack#1295]
### What happened
Using `AddRemoteFile` gives an `invalid_auth` error when used with a token with proper permissions.

### Expected behavior
Should add the remote file.

### Steps to reproduce
Create a new app, give appropriate permissions, attempt to call `AddRemoteFile`.

#### reproducible code
...

#### manifest.yaml
...

### Versions
- Go: 1.22.4
- slack-go/slack: latest

The fix for this is to remove this line: https://github.com/slack-go/slack/blob/master/remotefiles.go#L113

**Comment by lorenzoaiello** [comment]
Hi @jjjjeeffff ,

I tried to reproduce this locally and wasn't able to.

This is the code snippet I used:

```go
func main() {
	api := slack.New("xoxb-****")
	params := slack.RemoteFileParameters{
		Title:       "My File",
		ExternalID:  "my-file-123",
		ExternalURL: "https://raw.githubusercontent.com/slack-go/slack/master/README.md",
	}
	file, err := api.AddRemoteFile(params)
	if err != nil {
		fmt.Printf("%s\n", err)
		return
	}
	fmt.Printf("Name: %s, URL: %s\n", file.Name, file.URLPrivate)

	err = api.DeleteFile(file.ID)
	if err != nil {
		fmt.Printf("%s\n", err)
		return
	}
	fmt.Printf("File %s deleted successfully.\n", file.Name)
}
```

With a brand new application with the `remote_files:write` permission on my workspace.

Can you help me reproduce this so I try to figure out what might be going on?

**Comment by jjjjeeffff** [comment]
It still reproduces for me:
```2024/07/17 12:51:10 adding remote file: invalid_auth```

If I remove the line I mentioned above, the error goes away and the upload works. Very strange! My usage is very similar to yours, I do assign `params.PreviewImageReader` but I don't think that makes a difference.

Edit:
Ok I looked closer and actually setting the `PreviewImageReader` [_does_ make a difference](https://github.com/slack-go/slack/blob/master/remotefiles.go#L126-L127). By default, `remoteFileRequest` is used, but when `PreviewImageReader` is set it uses `postWithMultipartResponse`, which uses a func that [sets the token in query string](https://github.com/slack-go/slack/blob/master/misc.go#L72) (which Slack doesn't support here).

**Comment by lorenzoaiello** [comment]
For documentation purposes, I was able to reproduce the issue with this code snippet and also verified that the PR (#1117) fixes this issue and appears to be backwards compatible to the best I can tell:

```
	api := slack.New("YOUR_TOKEN_HERE")
	previewImage, err := os.Open("img.png")
	params := slack.RemoteFileParameters{
		Title:              "My File",
		ExternalID:         "my-file-123",
		ExternalURL:        "https://raw.githubusercontent.com/slack-go/slack/master/README.md",
		PreviewImage:       "img.png",
		PreviewImageReader: previewImage,
	}
	file, err := api.AddRemoteFileContext(context.Background(), params)
	if err != nil {
		fmt.Printf("%s\n", err)
		return
	}
	fmt.Printf("Name: %s, URL: %s\n", file.Name, file.URLPrivate)
```

**Comment by lorenzoaiello** [comment]
Resolved as part of [v0.14.0](https://github.com/slack-go/slack/releases/tag/v0.14.0).
